x-pack/plugins/infra/public/alerting/log_threshold/components/expression_editor/hooks/use_chart_preview_data.tsx

```diff
@@ -43,6 +46,55 @@ export const useChartPreviewData = ({
       cancelPreviousOn: 'creation',
       createPromise: async () => {
         setHasError(false);
+        if (isRatioRule(ruleParams.criteria)) {
+          const ratio = await Promise.all([
+            callGetChartPreviewDataAPI(
+              logViewReference,
+              http!.fetch,
+              { ...ruleParams, criteria: [...ruleParams.criteria[0]] },
+              buckets,
+              executionTimeRange
+            ),
+            callGetChartPreviewDataAPI(
+              logViewReference,
+              http!.fetch,
+              { ...ruleParams, criteria: [...ruleParams.criteria[1]] },
+              buckets,
+              executionTimeRange
+            ),
+          ]);
+          // The two array have the same length and the same time range.
+          let seriesQueryA = ratio[0].data.series[0].points;
+          let seriesQueryB = ratio[1].data.series[0].points;
+
+          // When groupBy and a filter is applied, return the ratio only for the filtered grouped-by
+          if (ruleParams.groupBy.length && filterSeriesByGroupName) {
+            seriesQueryA =
+              ratio[0].data.series.find((series) => series.id === filterSeriesByGroupName)
+                ?.points || [];
+
+            seriesQueryB =
+              ratio[1].data.series.find((series) => series.id === filterSeriesByGroupName)
+                ?.points || [];
+          }
+
+          const ratioPoints = [];
+          for (let index = 0; index < seriesQueryA.length; index++) {
+            const point = {
+              timestamp: seriesQueryA[index].timestamp,
+              value: 0,
+            };
+            // We follow the mathematics principle that dividing by 0 isn't possible,
+            if (seriesQueryA[index].value === 0 || seriesQueryB[index].value === 0) {
+              ratioPoints.push(point);
+            } else {
+              const ratioValue = seriesQueryA[index].value / seriesQueryB[index].value;
+              ratioPoints.push({ ...point, value: ratioValue });
+            }
+          }
+          const series = [{ id: 'ratio', points: ratioPoints }];
```

Comment:
**doubt**: should the id be set to `filterSeriesByGroupName`? In my understanding, the result will be the series for the passed group name, not the static `ratio` keyword.

Reply:
@tonyghiani, In the case of groupBy, you are right, the id should be `filterSeriesByGroupName` and not `ratio`

Reply:
updating it now

Reply:
Done ✅